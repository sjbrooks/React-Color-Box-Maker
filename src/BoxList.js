import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from 'uuid';

function BoxList() {
  const [boxes, setBoxes] = useState([]);

  // function to pass down to Box as a prop to remove the box upon click of a button
  function remove(box) {
    console.log('Made It into REMOVE in BoxList')
    console.log(`\n\n\n The value of box is `, box);
    let boxToRemoveID = box.id;
    console.log(`\n\n\n The value of box.id is `, box.id);
    setBoxes(boxes.filter(b => b.id !== boxToRemoveID));
  }

  // const remove = box => {

  // };

  const renderBoxes = () => {
    return (
      <div>
        {boxes.map(box => (
          <div>
            <Box key={box.id} id={box.id} backgroundColor={box.backgroundColor} width={box.width} height={box.height} remove={remove} />
          </div>
        ))}
      </div>
    );
  };
  // end renderBoxes

  /** Add new box to BoxList. */
  const addBox = box => {
    let newBox = { ...box, id: uuid() };
    setBoxes(boxes => [...boxes, newBox]);
  };
  // end addBox

  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      {renderBoxes()}
    </div>
  );
};
// end

export default BoxList;
