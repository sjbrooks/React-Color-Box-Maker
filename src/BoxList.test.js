import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList"

function addBox(boxList, height = "256", width = "256", backgroundColor = "orange") {
    const heightInput = boxList.getByLabelText("Height:");
    const widthInput = boxList.getByLabelText("Width:");
    const backgroundInput = boxList.getByLabelText("Color:");
  
    fireEvent.change(backgroundInput, { target: { value: backgroundColor } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Add a new box!");
    fireEvent.click(button);
  }
  

test('renders BoxList without crashing', () => {
    render(< BoxList />)
});


test("snapshot:  renders Box List Page each time", function(){
    const{asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
})

test("addBox adds box to BoxList ", function(){
    const boxList = render(<BoxList />)
    addBox(boxList);

    expect()
})






