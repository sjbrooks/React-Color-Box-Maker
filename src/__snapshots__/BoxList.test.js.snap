// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot:  renders Box List Page each time 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="backgroundColor"
      >
        Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="0"
      />
      <button>
        Add a new box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;

exports[`snapshot:  renders boxz each time 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="backgroundColor"
      >
        Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="0"
      />
      <button>
        Add a new box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;

exports[`snapshot:  renders each time 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="backgroundColor"
      >
        Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value="0"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value="0"
      />
      <button>
        Add a new box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;

exports[`snapshot:  renders same form each time 1`] = `
<DocumentFragment>
  <form>
    <label
      for="backgroundColor"
    >
      Color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      value=""
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      type="number"
      value="0"
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      type="number"
      value="0"
    />
    <button>
      Add a new box!
    </button>
  </form>
</DocumentFragment>
`;
